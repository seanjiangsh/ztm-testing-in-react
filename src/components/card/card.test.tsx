import { shallow } from "enzyme";

import Card, { CardProps } from "./card.component";

const cardProps: CardProps = {
  monster: { id: 1, name: "Test Monster", email: "whatever@gmail.com" },
};
test("expect to render Card component", () => {
  const comp = shallow(<Card {...cardProps} />);
  expect.assertions(1);
  expect(comp.length).toEqual(1);
});
