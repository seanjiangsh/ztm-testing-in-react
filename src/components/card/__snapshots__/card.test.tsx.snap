// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Card component 1`] = `
<div
  className="card-container"
>
  <img
    alt="monster Test Monster"
    src="https://robohash.org/1?set=set2&size=180x180"
  />
  <h2>
    Test Monster
  </h2>
  <p>
    whatever@gmail.com
  </p>
</div>
`;
